prototype1/src/App.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef } from "react"
+import "./App.css"
+import { data } from "./data"
+import * as d3 from "d3"
+import { useWindowSize } from "react-use"
+
+let { children } = data
+let id = (x: any) => x
+
+function useMount(fn: Function) {
+  useEffect(() => fn(), [])
+}
+
+const padding = 30
+const textSpacingFromCircle = 5
+
+function App() {
+  let svgRef = useRef<SVGSVGElement>(null)
+  let rootRef =
+    useRef<d3.Selection<SVGGElement, unknown, null, undefined>>(null)
+  let sizeProps = useWindowSize(0, 0)
+  let paddedSizeProps = {
+    height: sizeProps.height - padding * 2,
+    width: sizeProps.width - padding * 2,
+  }
+
+  useEffect(() => {
+    let svg = d3.select(svgRef.current)
+
+    let root = d3
+      .hierarchy(data)
+      // TODO: Derrive size from file/folder size
+      .sum(() => 10)
+      .sort((a, b) => b.value - a.value)
+
+    let partition = d3
+      .pack()
+      .size([paddedSizeProps.width, paddedSizeProps.height])
+      .padding(padding)
+
+    partition(root)
+
+    rootRef.current = svg.append("g")
```

Comment:
This error can be removed by setting line 20 to
```typescript
useRef<d3.Selection<SVGGElement, unknown, null, undefined> | null>(null)
```

Reply:
The ref corresponds to the DOM SVG element, so this type anotation is not valid.